File: core/src/main/java/com/simple/castle/core/ModelSend.java
package com.simple.castle.core;

import com.badlogic.gdx.math.Matrix4;

import java.io.Serializable;

public final class ModelSend implements Serializable {
    private final String id;
    private final Matrix4 matrix4;

    public ModelSend(String id, Matrix4 matrix4) {
        this.id = id;
        this.matrix4 = new Matrix4(matrix4);
    }

    public String getId() {
        return id;
    }

    public Matrix4 getMatrix4() {
        return matrix4;
    }

    @Override
    public String toString() {
        return "ModelSend{" +
                "id='" + id + '\'' +
                ", matrix4=" + matrix4 +
                '}';
    }
}

File: core/src/main/java/com/simple/castle/core/kt/api/ModelSend.kt
package com.simple.castle.core.kt.api

import com.badlogic.gdx.math.Matrix4
import java.io.Serializable

class ModelSend(val id: String?, matrix4: Matrix4?) : Serializable {
    val matrix4: Matrix4 = Matrix4(matrix4)

    override fun toString(): String {
        return "ModelSend{" +
                "id='" + id + '\'' +
                ", matrix4=" + matrix4 +
                '}'
    }

}